src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,42 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname ) {
+        _cleanup_free_ char *promote_secondaries_sysctl = NULL,*promote_secondaries_path = NULL;
+
+        promote_secondaries_path = strjoin("net/ipv4/conf/", ifname, "/promote_secondaries");
+        if (sysctl_read(promote_secondaries_path, &promote_secondaries_sysctl) >= 0) {
+                unsigned val;
+
+                truncate_nl(promote_secondaries_sysctl);
+                if (safe_atou(promote_secondaries_sysctl, &val) >= 0) {
+                        if (val > 0)
+                                return true;
+                }
+        }
+
+        return false;
+}
+
+int dhcp4_check_promote_secondaries(Link *link) {
+        assert(link);
+        assert(link->network);
+        assert(link->network->dhcp & ADDRESS_FAMILY_IPV4);
+
+        if (!(check_promote_secondaries("all") ||check_promote_secondaries("default") || check_promote_secondaries(link->ifname))) {
```

Comment:
please add a comment what this is supposed to do, i don't grok it

Reply:
I added a comment here and to the function itself, I hope this helps explaining it somewhat better.